conanfile.py

```diff
@@ -47,21 +48,39 @@ def configure(self):
         self.options["freetype"].shared = self.settings.os == "Android"
 
         # SDL_image and Qt depend on it, in iOS both are static
-        # Enable static libpng due to https://github.com/conan-io/conan-center-index/issues/15440,
-        # which leads to VCMI crashes of MinGW
-        self.options["libpng"].shared = not (self.settings.os == "Windows" and cross_building(self)) and self.settings.os != "iOS"
+        self.options["libpng"].shared = self.settings.os != "iOS"
```

Comment:
not sure which limitations you mean. Currently the only real limitations are:
- Qt for iOS can only built as static
- oneTBB must be shared (their heavy recommendation)

I follow a simple rule: if a dependency is needed by more than 2 "separate" targets (e.g. an app and a shared library), then the dependency should be built as shared, otherwise static should be fine.